### PR TITLE
Deprecate unused functions in cbook.

### PR DESCRIPTION
**Pull request by anntzer**
I think most of the deprecations I propose should be pretty uncontroversial, but feel free to argue otherwise.  Let's see whether we can work on reducing the old API...

### CODE REVIEW

File: lib/matplotlib/cbook.py [hunk 1]
```diff
@@ -2569,6 +2607,7 @@ def normalize_kwargs(kw, alias_mapping=None, required=(), forbidden=(),
     return ret
 
 
+@deprecated('2.0')
```
Comment (tacaswell) [inline lib/matplotlib/cbook.py]: This one should be use in the pandas support code...

**Comment by tacaswell** [comment]
Given that we just did an rc without these, I think they should target 2.1 instead of 2.0.

**Comment by NelleV** [comment]
I actually think that everything currently tagged 2.0.1 that isn't a major bug should target 2.1

**Comment by dopplershift** [comment]
@NelleV Agree, except scratch the work 'major'. *Any* bugfix should target the next bugfix release.

**Comment by anntzer** [comment]
Sounds fair.

**Comment by NelleV** [comment]
This is long overdue… Thanks for tackling this @anntzer !

**Pull request merged**